Tasks/Common/webdeployment-common/msdeployutility.ts

```diff
@@ -98,45 +99,67 @@ export function getMSDeployCmdArgs(webAppPackage: string, webAppName: string, pu
+        isParamFilePresentBackup = await isMSDeployPackage(webAppPackage);
     }
-    catch(error) {
-        throw Error(error);
+    catch (error) {
+        isParamFilePresentBackup = false;
+        tl.debug("an error occurred in the function isMSDeployPackage");
     }
-    finally {
-        fs.fsyncSync(fd);
-        fs.closeSync(fd);
+    var msDeployCheckParamFileCmdArgs = "-verb:getParameters -source:package=\'" + webAppPackage + "\'";
+    var res =  tl.execSync("msdeploy", msDeployCheckParamFileCmdArgs, <any>{ failOnStdErr: true });
+
+    if((res.error && res.error.message.length > 0 ) || (res.stderr && res.stderr.length > 0)) {
+        tl.debug("[COMMAND FAILED]: msdeploy " + msDeployCheckParamFileCmdArgs);
+        if(res.error) {
+            tl.debug("res.error: " + res.error.message);
+        }
+        if(res.stderr) {
+            tl.debug("res.stderr: " + res.stderr);
+        }
+        isParamFilePresent = isParamFilePresentBackup;
+        tl.debug("Is parameter file present in web package : " + isParamFilePresent);
+        return isParamFilePresent;
     }
 
-    var paramContentXML = fs.readFileSync(parameterFile).toString();
-    paramContentXML = paramContentXML.slice(paramContentXML.indexOf('\n') + 1, paramContentXML.length);
-    var isParamFilePresent = false;
+    var paramContentXML = res.stdout;
 
     await parseString(paramContentXML, (error, result) => {
         if(error) {
-            throw new Error(error);
+            isParamFilePresent = isParamFilePresentBackup;
         }
         if(result != null && result['output'] != null && result['output']['parameters'] != null) {
             if(result['output']['parameters'][0] ) {
                 isParamFilePresent = true;
             }
         }
         else {
-            tl.warning("Unable to parse the content of parameterFile: "+parameterFile);
+            isParamFilePresent = isParamFilePresentBackup;
+            tl.warning("Parameter file is not well formed");
             tl.debug("Parameter File Content is:");
             tl.debug(paramContentXML);
         }
-
     });
 
     tl.debug("Is parameter file present in web package : " + isParamFilePresent);
-    tl.rmRF(parameterFile, true);
+    return isParamFilePresent;
+}
+
+/**
+ * Check whether the package contains parameter.xml file
+ * @param   webAppPackage   web deploy package
+ * @returns boolean
+ */
+async function isMSDeployPackage(webAppPackage: string ) {
+    var isParamFilePresent = false;
+    var pacakgeComponent = await zipUtility.getArchivedEntries(webAppPackage);
+    if (((pacakgeComponent["entries"].indexOf("parameters.xml") > -1) || (pacakgeComponent["entries"].indexOf("Parameters.xml") > -1)) && 
+    ((pacakgeComponent["entries"].indexOf("systeminfo.xml") > -1) || (pacakgeComponent["entries"].indexOf("systemInfo.xml") > -1))) {
+        isParamFilePresent = true;
+    }
+    tl.debug("Is parameter file present in web package : " + isParamFilePresent);
```

Comment:
This debug line is not relevant for this function. This function should print if this is valid msdeploy package or not. Parameter file presence should be printed in checkparamfile function.